Create link.product block (#1)
vtex-apps/store-link: docs/README.md

> # VTEX Store Link

Reviewer: # Store Link

This repository provides some blocks that can help you creating links in a store.

![image](https://user-images.githubusercontent.com/8517023/73387868-f1b36f80-42af-11ea-8e24-3045d2c819b4.png)

## Configuration

1. Add `store-link` app to your theme's dependencies in the `manifest.json`, for example:

```jsonc
{
  "vtex.store-link": "0.x"
}
```

> 2. Now you can use the blocks exported by the app. In the following example you can see the usage of the `link.product` inside of a quick view component:

Reviewer: I would add a list of all blocks exported by the app. Is it possible? Although we have an example on how to use the `link.product` block, as user I dont know which blocks can also be used once I dont know which blocks actually were exported with the `store-link` app. What do you think?

Reviewer: For example: 

```jsonc
{
  "vtex.store-link": "0.x"
}
```
Now, you are able to use all blocks exported  by the `store-link` app. Check out the full list below: 

| Block name | Description | 
| --------- | -------- | 
| `link.product`   | oeeubdan | 
| `link`    | njbduwd | 

2. Based on the exported list, choose the desired block and declare it in the block that will host the link. Below you can find an example of a `link.product` block being used in the Quick View component:

Reviewer: What about this Quick View component? I dont know what this is :(

Author: The quickview is the main use case of `modal-layout`. It isn't an app tho. https://github.com/vtex-apps/store-theme/pull/195

Author: > What about this Quick View component? I dont know what this is :(

The quickview is the main use case of `modal-layout`. It isn't an app tho. https://github.com/vtex-apps/store-theme/pull/195

```jsonc
{
  "link.product#product-page": {
    "props": {
      "href": "/{slug}/p",

>       "label": "Mais detalhes >"

Reviewer: It would be better if the example was fully in english. What do you think?

    }
  },
  "product-summary.shelf": {
    "children": [
      "product-summary-image",
      "product-summary-name",
      "product-rating-inline",
      "product-summary-space",
      "product-summary-price",
      "link.product#product-page
    ]
  },
}
```

> You can see that there is a `{slug}` placeholder being passed to the `href` prop. When rendered, the placeholder will be replaced with the respective value of the closest product context and generate a link like `/everyday-necessaire/p`. However, keep in mind that in order for this work, you have to place the `link.product` block inside of a another block that provides a product context, such as the [`ProductSummary`](https://vtex.io/docs/components/product/vtex.product-summary).

Reviewer: :warning: Note that there is a `{slug}` **placeholder** being passed onto the `href` prop in the example above. When rendered, this placeholder will be overwritten by the value accrued from the closest product context, generating a link like `/everyday-necessaire/p`. Therefore, remember that in order for this format to work you have to place the `link.product` block inside of a another block that provides a product context, such as the [`ProductSummary`](https://vtex.io/docs/components/product/vtex.product-summary).

> ### Props

Reviewer: No need to create a section for props. They can be in the ## Configuration section :)

> The `link.product` and `link` blocks has the same props.

Reviewer: The `link.product` and `link` blocks have the same props:

(If there are any other block, we can say "All blocks exported by `store-link` share the same props:")

| Prop name | Type     | Description                             | Default value |
| --------- | -------- | --------------------------------------- | ------------- |
| `label`   | `string` | The label that you be shown to the user | `undefined`   |

> | `href`    | `string` | The actual link                         | `'#'`         |

Reviewer: `label` -> Link text
`link` -> Link URL

Reviewer: I didnt understand what `#` is :(

Reviewer: `undefined` is for a default value that does not exist? We can apply a default description for that. Previously, I was using just a -, but now Im afraid that people misunderstood this - as a default value haha. We could use a  sentence more clarified, such as: "No default value is defined"

What do you think?

Author: > `undefined` is for a default value that does not exist? We can apply a default description for that. Previously, I was using just a -, but now Im afraid that people misunderstood this - as a default value haha. We could use a sentence more clarified, such as: "No default value is defined"
> 
> What do you think?

`undefined` is a value in javascript that means exactly what you said, so I think is ok to let `undefined`. 

The default value ` #` means that it will add just a hash to the url if you click, which means that nothing will happen, is kinda a default value of links in general

> ### Available variables

Reviewer: When creating a Link URL for your `link.product` block, use the variables listed below. With them, you will be able to structure any desired URL for your store, such as a link to a given product department (`/{department}`).

Reviewer: No need for a new section for available variables, I believe

> #### link.product

Reviewer: No need for this section too

> | Value          | Description                                                                                    |

Reviewer: Variable value | Description

| -------------- | ---------------------------------------------------------------------------------------------- |

> | `'slug'`       | The slug of the product                                                                        |

Reviewer: Product slug

Reviewer: What does it mean?

Reviewer: I believe it would be better if we give more details about what should be filled out in `slug`

> | `'skuId'`      | The id of the current sku                                                                      |

Reviewer: Current SKU ID

Reviewer: What does it mean? "current"?

> | `'department'` | The department of the product                                                                  |

Reviewer: Product department

> | `'category1'`  | The most general category of the product, the last available category is the most specific one |

Reviewer: Highest level category in the category tree

> | `'category2'`  | The second most general category                                                               |

Reviewer: Second highest level category

> | `'category3'`  | -                                                                                              |

Reviewer: Third highest level category

> | `'category4'`  | -                                                                                              |

Reviewer: Fourth highest level category

> | `'productId'`  | The id of the product                                                                          |

Reviewer: Product ID

> | `'brand'`      | Brand name                                                                                     |

Reviewer: Name of the product brand

> | `'brandId'`    | Brand id                                                                                       |

Reviewer: ID of the product brand

> With these variables you can form any url that you want, you can do something like `/{department}` and create a link to go to the department of the product

Reviewer: Already written above :) 

No customizations are available for this app?

# VTEX Store Link

> This repository has the blocks that you should use to create link in the store

Reviewer 2: ```suggestion
This repository provides some blocks that can help you creating links in a store.
```

![image](https://user-images.githubusercontent.com/8517023/73387868-f1b36f80-42af-11ea-8e24-3045d2c819b4.png)

## Configuration

> 1. Import the store link's app to your theme's dependencies in the `manifest.json`, for example:

Reviewer 2: ```suggestion
1. Add `store-link` app to your theme's dependencies in the `manifest.json`, for example:
```

```jsonc
{
  "vtex.store-link": "0.x"
}
```

> 2. Now you can use the block that the `store-link` app export. In the following example you can see the usage of the `store-link.product` inside of a quickview component:

Reviewer 2: ```suggestion
2. Now you can use the blocks exported by the app. In the following example you can see the usage of the `store-link.product` inside of a quick view component:
```

```jsonc
{
  "store-link.product#quickview": {
    "props": {
      "href": "/{slug}/p",

>       "label": "Mais detalhes >",

Reviewer 3: It would be nice to have an english example here

      "blockClass": "quickview"
    }
  },
  "flex-layout.col#quickview-product-information": {
    "children": [
      "vtex.store-components:product-name",
      "product-rating-summary",
      "product-price#quickview-price",
      "product-separator",
      "product-identifier.product",
      "store-link.product#quickview"
    ],
    "props": {
      "preventVerticalStretch": true,
      "rowGap": 0,
      "width": "calc(50% - 0.375rem)"
    }
  },
  "product-price#quickview-price": {
    "props": {
      "showInstallments": true,
      "showSavings": true
    }
  }
}
```

> You can see that it is passing the href prop with a `{slug}` thing, that will take the value of the product context that it is inserted and use it to generate a link like `/everyday-necessaire/p`. But you need to have in mind that in order to this work you have to put the `store-link.product` block inside of a another block that provides a product context. In this case the [ProductSummary](https://vtex.io/docs/components/product/vtex.product-summary) provides a product context and the `modal-layout` is included the `product-summary` and the `store-link.product` is included in the `modal-layout` that is why it has the product context.

Reviewer 4: ```suggestion
You can see that it is passing the href prop with a `{slug}` variable, that will take the value of the product context that it is inserted and use it to generate a link like `/everyday-necessaire/p`. But you need to have in mind that in order to this work you have to put the `store-link.product` block inside of another block that provides a product context. In this case, the [ProductSummary](https://vtex.io/docs/components/product/vtex.product-summary) provides a product context and the `modal-layout` containing `store-link.product`  is included in the `product-summary`, that is why it has the product context.
```

Reviewer 4: Beware that there is no `modal-layout` in the example above

> You can see that it is passing the href prop with a `{slug}` thing, that will take the value of the product context that it is inserted and use it to generate a link like `/everyday-necessaire/p`. But you need to have in mind that in order to this work you have to put the `store-link.product` block inside of a another block that provides a product context. In this case the [ProductSummary](https://vtex.io/docs/components/product/vtex.product-summary) provides a product context and the `modal-layout` is included the `product-summary` and the `store-link.product` is included in the `modal-layout` that is why it has the product context.

Reviewer 2: ```suggestion
You can see that there is a `{slug}` placeholder being passed to the `href` prop. When rendered, the placeholder will be replaced with the respective value of the closest product context and generate a link like `/everyday-necessaire/p`. However, keep in mind that in order for this work, you have to place the `store-link.product` block inside of a another block that provides a product context, such as the [`ProductSummary`](https://vtex.io/docs/components/product/vtex.product-summary) or the `modal-layout`, which already includes a `product-summary` block.
```

Author: > 

"which already includes a `product-summary` block", the `modal-layout` is included in the `product-summary`

### store-link.product

> The props of this block are very simple

Reviewer 2: Needed? Why not `#### Props`?

| Prop name | Type     | Description                             | Default value |
| --------- | -------- | --------------------------------------- | ------------- |
| `label`   | `string` | The label that you be shown to the user | `undefined`   |
| `href`    | `string` | The actual link                         | `'#'`         |

> #### Available variables

Reviewer 2: ```suggestion
#### Available placeholders
```

# Store Link

This repository provides some blocks that can help you creating links in a store.

![image](https://user-images.githubusercontent.com/8517023/73387868-f1b36f80-42af-11ea-8e24-3045d2c819b4.png)

## Configuration

1. Add `store-link` app to your theme's dependencies in the `manifest.json`, for example:

```jsonc
{
  "vtex.store-link": "0.x"
}
```

> Now, you are able to use all blocks exported by the `store-link` app. Checkout the full list below:

Reviewer: Check out

| Block name     | Description                                           |
| -------------- | ----------------------------------------------------- |

> | `link.product` | A link a component that consumes the product context. |

Reviewer: A link that consumes Product context

| `link`         | A normal link that doesn't consume any context.       |

> 2. Based on the exported list, choose the desired block and declare it in the block that will host the link. Below you can find an example of a link.product block being used in the [product-summary](https://vtex.io/docs/components/product/vtex.product-summary), but you can see a more complex example in the [Quick View](https://github.com/vtex-apps/modal-layout/blob/master/docs/README.md#modal-layout) that uses the `modal-layout` blocks:

Reviewer: 2. Based on the exported list, choose the desired block and declare it in the block that will host the link. Find below an example of a `link.product` being used in the [product-summary](https://vtex.io/docs/components/product/vtex.product-summary block):

```jsonc
{
  "link.product#product-page": {
    "props": {
      "href": "/{slug}/p",
      "label": "More details >"
    }
  },
  "product-summary.shelf": {
    "children": [
      "product-summary-image",
      "product-summary-name",
      "product-rating-inline",
      "product-summary-space",
      "product-summary-price",
      "link.product#product-page
    ]
  },
}
```

> 

Reviewer: A more complex example with `modal-layout` blocks can be found in the [Quick View](https://github.com/vtex-apps/modal-layout/blob/master/docs/README.md#modal-layout) documentation.

}
```

A more complex example with `modal-layout` blocks can be found in the [Quick View](https://github.com/vtex-apps/modal-layout/blob/master/docs/README.md#modal-layout) example at the `modal-layout` documentation.

:warning: Note that there is a `{slug}` **placeholder** being passed onto the `href` prop in the example above. When rendered, this placeholder will be overwritten by the value accrued from the closest product context, generating a link like `/everyday-necessaire/p`. Therefore, remember that in order for this format to work you have to place the `link.product` block inside of a another block that provides a product context, such as the [`ProductSummary`](https://vtex.io/docs/components/product/vtex.product-summary).

## Props

> All blockx exported by `store-link` share the same props:

Reviewer 5: ```suggestion
All blocks exported by `store-link` share the same props:
```
